indexer/src/main.ts

```diff
@@ -73,206 +79,238 @@ export default async function transform({
-        encodedReceipt,
-      }) => {
-        // Add the transaction to the transaction trie.
-        await transactionTrie.put(encodedTransactionIndex, encodedTransaction);
-        // Add the receipt to the receipt trie.
-        await receiptTrie.put(encodedTransactionIndex, encodedReceipt);
-      },
-    ),
-  );
+  const { transactionTrie, receiptTrie } = await computeBlooms(trieData);
 
   // Sort the cumulative gas uses by descending transaction index.
   cumulativeGasUsages.reverse();
 
-  (transactions ?? []).forEach(({ transaction, receipt }) => {
-    if (isRevertedWithOutOfResources(receipt)) {
-      // Can be false if the transaction is not related to a specific instance of the Kakarot contract.
-      // This is typically the case if there are multiple Kakarot contracts on the same chain.
-      const isKakarotTx = isKakarotTransaction(transaction);
-      if (!isKakarotTx) {
-        return;
-      }
-
-      const ethTx = toEthTx({
-        transaction,
-        receipt,
-        blockNumber,
-        blockHash,
-        isPendingBlock,
-      });
-      if (!ethTx) {
-        return;
-      }
-
-      // Get the cumulative gas used for the reverted transaction.
-      // Example:
-      // const cumulativeGasUsages = [300n, undefined, undefined, 200n, undefined, 100n, undefined, undefined, 10n, undefined];
-      // const ethTx = { transactionIndex: 5 };
-      // const revertedTransactionCumulativeGasUsed = 100n;
-      const len = cumulativeGasUsages.length;
-      const revertedTransactionCumulativeGasUsed =
-        cumulativeGasUsages.find((gas, i) => {
-          return (
-            Number(ethTx.transactionIndex) >= len - 1 - i && gas !== undefined
-          );
-        }) ?? 0n;
-
-      const ethReceipt = toRevertedOutOfResourcesReceipt({
-        transaction: ethTx,
-        blockNumber,
-        blockHash,
-        cumulativeGasUsed: revertedTransactionCumulativeGasUsed,
-        isPendingBlock,
-      });
-
-      store.push({ collection: "transactions", data: { tx: ethTx } });
-      store.push({
-        collection: "receipts",
-        data: {
-          receipt: ethReceipt,
-        },
-      });
-    }
-  });
+  const processedTransactions = processTransactions(
+    transactions,
+    blockInfo,
+    cumulativeGasUsages,
+  );
+  updateStoreWithTransactions(store, processedTransactions);
 
   const ethHeader = await toEthHeader({
     header: header,
     gasUsed: cumulativeGasUsed,
     logsBloom: blockLogsBloom,
     receiptRoot: receiptTrie.root(),
     transactionRoot: transactionTrie.root(),
-    blockNumber,
-    blockHash,
-    isPendingBlock,
+    ...blockInfo,
   });
   store.push({
-    collection: "headers",
+    collection: Collection.Headers,
     data: { header: ethHeader },
   });
 
   return store;
 }
+
+function createBlockInfo(header: BlockHeader): BlockInfo {
+  const blockNumber = padString(toHexString(header.blockNumber), 8);
+  const blockHash = padString(header.blockHash, 32);
```

Comment:
how does this differ from native `.padStart` or `.padEnd` ?

Reply:
padString adds a "0x" prefix and handles default values, ensuring proper hexadecimal formatting, while native .padStart and .padEnd only pad strings without any additional logic